assignments/prototype-refactor.js

```diff
@@ -7,3 +7,178 @@ Prototype Refactor
+  * destroy() // prototype method that returns: `${this.name} was removed from the game.`
+*/
+// function GameObject (createdAt,name, dimensions) {
+//     this.createdAt = createdAt;
+//     this.name = name;
+//     this.dimensions = dimensions;
+//   }
+//   GameObject.prototype.destroy = function () {
+//     return `${this.name} was removed from the game.`
+//   }
+
+  class GameObject {
+      constructor ({createdAt,name, dimensions}) {
+        this.createdAt = createdAt;
+        this.name = name;
+        this.dimensions = dimensions;
+      }
+
+      destroy () {
+        return `${this.name} was removed from the game.`
+      }
+  }
+  /*
+    === CharacterStats ===
+    * healthPoints
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+//   function CharacterStats (createdAt,name, dimensions,healthPoints) {
+//     GameObject.call(this,createdAt,name, dimensions)
+//     this.healthPoints =healthPoints
+//   }
+//   CharacterStats.prototype = Object.create(GameObject.prototype)
+//   CharacterStats.prototype.takeDamage = function() {
+//     return `${this.name} took damage`
+//   }
+
+  class CharacterStats extends GameObject {
+      constructor ({createdAt,name, dimensions,healthPoints}) {
+          super({createdAt,name, dimensions});
+          this.healthPoints =healthPoints;
+      }
+
+      takeDamage () {
+        return `${this.name} took damage`
+      }
+  }
+  
+  /*
+    === Humanoid (Having an appearance or character resembling that of a human.) ===
+    * team
+    * weapons
+    * language
+    * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+    * should inherit destroy() from GameObject through CharacterStats
+    * should inherit takeDamage() from CharacterStats
+  */
+//   function Humanoid (attributes) {
+//     CharacterStats.call(this,attributes);
+//     this.team = attributes.team;
+//     this.weapons = attributes.weapons;
+//     this.language = attributes.language;
+//     this.name = attributes.name;
+//     this.dimensions = attributes.dimensions;
+//     this.createdAt = attributes.createdAt;
+//     this.healthPoints = attributes.healthPoints;
+//   }
+//   Humanoid.prototype = Object.create(CharacterStats.prototype)
+//   Humanoid.prototype.greet = function () {
+//     return `${this.name} offers a greeting in ${this.language}.`
+//   }
+  class Humanoid extends CharacterStats {
+      constructor({team, weapons, language, name, dimensions, 
+        createdAt, healthPoints}) {
+        super({createdAt,name, dimensions,healthPoints})
+        this.team = team;
+        this.weapons = weapons;
+        this.language = language;
+        this.name = name;
+        this.dimensions = dimensions;
+        this.createdAt = createdAt;
+        this.healthPoints = healthPoints;
```

Comment:
As the last day project you have too many property declarations in this class. 
You are already inheriting those 4 last properties from GameObject class and CharacterStats class, so you don't need to declare them again.